src/Riok.Mapperly/Descriptors/MappingBuilder/UserMethodMappingBuilder.cs

```diff
@@ -14,15 +14,19 @@ public static class UserMethodMappingBuilder
         // extract user implemented and user defined mappings from mapper
         foreach (var methodSymbol in ExtractMethods(mapperSymbol))
         {
-            var mapping = BuilderUserDefinedMapping(ctx, methodSymbol) ?? BuildUserImplementedMapping(methodSymbol);
+            var mapping = BuilderUserDefinedMapping(ctx, methodSymbol) ?? BuildUserImplementedMapping(methodSymbol, false);
             if (mapping != null)
                 yield return mapping;
         }
 
         // extract user implemented mappings from base methods
         foreach (var method in ExtractBaseMethods(ctx.Compilation.ObjectType, mapperSymbol))
         {
-            var mapping = BuildUserImplementedMapping(method);
+            // partial method declarations are allowed vor base classes,
```

Comment:
Change this to
> Partial method declarations are allowed for base classes,
> but still treated as user implemented methods,
> since the user should provide an implementation elsewhere.
> This is the case if a partial mapper class is extended.

Reply:
Improved.